lab_1_keywords_tfidf/main.py

```diff
@@ -3,108 +3,164 @@
 Extract keywords based on frequency related metrics
 """
 from typing import Optional, Union
+import math
 
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
-
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Removes punctuation, casts to lowercase, splits into tokens
+       Parameters:
+       text (str): Original text
+       Returns:
+       list[str]: A sequence of lowercase tokens with no punctuation
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(text, str):
+        target_text = text.lower().strip()
+        symbols = """!()-[]{};?@#$%:'",.^&;*_><"""
+        for symbol in target_text:
+            if symbol in symbols:
+                target_text = target_text.replace(symbol, '')
+        tokens = target_text.split()
+        for word in tokens:
+            if word == "' '":
+                tokens.remove(word)
+        return tokens
+    return None
 
 
 def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
-    Excludes stop words from the token sequence
-
-    Parameters:
-    tokens (List[str]): Original token sequence
-    stop_words (List[str]: Tokens to exclude
-
-    Returns:
-    List[str]: Token sequence that does not include stop words
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Excludes stop words from the token sequence
+       Parameters:
+       tokens (List[str]): Original token sequence
+       stop_words (List[str]: Tokens to exclude
+       Returns:
+       List[str]: Token sequence that does not include stop words
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(tokens, list) and isinstance(stop_words, list):
```

Comment:
revert the condition and return None immediately

Reply:
done